factory.js

```diff
@@ -0,0 +1,45 @@
+const states = require('./states.js');
+
+function factory(input) {
+  let newState;
+  switch (input) {
+    case '{http://hl7.org/fhir}Encounter':
+      newState = new states.EncounterState('Encounter', 'encounter_class', ['codes']);
+      break;
+    case '{http://hl7.org/fhir}Condition':
+      newState = new states.ConditionOnsetState('Condition', 'target_encounter', ['codes']);
+      break;
+    case '{http://hl7.org/fhir}AllergyIntolerance':
+      newState = new states.AllergyOnsetState('AllergyIntolerance', 'target_encounter', ['codes']);
+      break;
+    case '{http://hl7.org/fhir}Medication':
+      newState = new states.MedicationOrderState('Medication', ['codes']);
+      break;
+    case '{http://hl7.org/fhir}CarePlan':
+      newState = new states.CarePlanStartState('CarePlan', ['codes']);
+      break;
+    case '{http://hl7.org/fhir}Procedure':
+      newState = new states.ProcedureState('Procedure', ['codes'], 'duration');
```

Comment:
One suggestion here, duration needs to either be an object or `undefined`. Let's leave duration out of this for now. So I would just delete the argument here, and see my corresponding comment in the ProcedureState class for how to handle it

```suggestion
      newState = new states.ProcedureState('Procedure', ['codes']);
```